### PR TITLE
Feature/auth housekeeping

### PR DESCRIPTION
**Pull request by dblane-digicatapult**
Adds token creation script
Health Checks and Kustomization/HelmRelease dependency management improved significantly.

### CODE REVIEW

File: clusters/kind-cluster/infra/ory/oathkeeper/configmap.yaml [hunk 1]
```diff
@@ -54,4 +54,4 @@ data:
       api:
         enabled: false
     maester:
-      enabled: true
+      enabled: false
```
Comment (mattdean-digicatapult) [inline clusters/kind-cluster/infra/ory/oathkeeper/configmap.yaml:57]: enabled `false`?

File: docs/getting-started.md [hunk 2]
```diff
@@ -31,6 +32,19 @@ Run
 
 Once this has completed you will have a functioning flux cluster.
 
+## Getting an API Token
+
+We use a combination of Ory [Hydra](https://www.ory.sh/docs/hydra) and Ory [Oathkeeper](https://www.ory.sh/docs/oathkeeper) for creating and storing OAuth2 access tokens which are in turn used to access the various APIs used in the dscp project.
```
Comment (mattdean-digicatapult) [inline docs/getting-started.md:37]: We should probably mention dependencies here

File: scripts/get-hydra-token.sh [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    kubectl cluster-info --context $context &> /dev/null
+    local ret=$?
+    if [ "$ret" -ne 0 ]; then
+        printf "NOT OK\nError accessing kind cluster $context. Have you created the kind cluster?\n"
+        exit 1
+    fi
+    printf "OK\nCluster $context exists\n"
+
+    printf "Checking if $namespace/postgresql is currently ready..."
+    kubectl wait -l statefulset.kubernetes.io/pod-name=pg-hydra-postgresql-0 --for=condition=ready pod -n $namespace  --timeout=300s &> /dev/null
```
Comment (mattdean-digicatapult) [inline scripts/get-hydra-token.sh:90]: Will the pod always be called this given the option switches?

**Review by mattdean-digicatapult — COMMENTED** [review]
A few comments/questions

File: clusters/kind-cluster/infra/ory/oathkeeper/configmap.yaml (hunk 1 above)
Comment (dblane-digicatapult) [reply clusters/kind-cluster/infra/ory/oathkeeper/configmap.yaml:57]: yes as I am now installing oathkeeper-maester as a standalone helm chart.

File: docs/getting-started.md (hunk 2 above)
Comment (dblane-digicatapult) [reply docs/getting-started.md:37]: It is mentioned on line 16.

Comment (mattdean-digicatapult) [reply docs/getting-started.md:37]: 🤦‍♂️

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**